src/streams/CreateStreamScreen.js

```diff
@@ -20,13 +22,19 @@ type Props = $ReadOnly<{|
 export default function CreateStreamScreen(props: Props): Node {
   const auth = useSelector(getAuth);
   const ownEmail = useSelector(getOwnEmail);
+  const getStreams = useSelector(getStreamsByName);
+  const allStreams = Array.from(getStreams.keys());
 
   const handleComplete = useCallback(
     (name: string, description: string, isPrivate: boolean) => {
-      api.createStream(auth, name, description, [ownEmail], isPrivate);
-      NavigationService.dispatch(navigateBack());
+      if (allStreams.includes(name)) {
```

Comment:
There's an edge case that I'm not sure how best to handle. I think it shouldn't block progress on the PR, but it'd be good to write it down in a code comment here, so at least the behavior is clear.

It's possible that a stream exists but the client doesn't know about it, for example, if it's a private stream that the user can't access.

Maybe something like

```js
      // This will miss existing streams that the client can't know about;
      // for example, a private stream the user can't access.
```

just above the `if` line.

Reply:
> I think it shouldn't block progress on the PR

Ah, I take that back; Greg has suggested something to do about this: https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/.23M5199/near/1364442

Reply:
Sure @chrisbobbe!, Thanks for thinking about this edge case. I think we forgot to discuss this edge case in our discussion. I'll add it.
Thanks 😊